### PR TITLE
New packages py-astrovascpy, py-vascpy, py-libsonata, and py-pytest-mpi

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-astrovascpy/package.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-seaborn", type=("build", "run"))
+    depends_on("py-tables", type=("build", "run"))
+    depends_on("py-tqdm", type=("build", "run"))
+    depends_on("py-trimesh", type=("build", "run"))
+    depends_on("py-vascpy", type=("build", "run"))
+    depends_on("vtk+python", type=("build", "run"), when="+vtk")
+
+    depends_on("py-pytest", type="test")
+    depends_on("py-pytest-mpi", type="test")
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/py-astrovascpy/package.py:54]: @tristan0x I find it highly unlikely that such large package these work with all versions of those packages. For example, https://github.com/BlueBrain/AstroVascPy/blob/main/pyproject.toml already states minimum versions of `setuptools` and `setuptools_scm`.
https://github.com/BlueBrain/AstroVascPy/blob/main/setup.py has `python_requires=">=3.10",`

If you can, please contact the upstream project and ask them for the minimum versions that they test with for their dependencies and ask them to add them to their requirements. I did not see that they declare this full requirements and supported Python versions in their pyproject.toml, which they should really do.

I guess, if they are not responsive, you may add the minimum versions that you built or tested with, add them as minimum versions to the depends on and add a comment that these are what have been tested by you.

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    version("0.1.20", sha256="f2e63a26dcf045bb7424e236e12f9b9ef8c46c356e953d26555f313f5763651e")
+    version("0.1.18", sha256="1b13728cbffbcb74cc864c7bc186e568945e25c260f732ffa872437c081fd9b4")
+    version("0.1.17", sha256="d122dd9fce82c8ce1621746ed9029e7db0d319fb94f6a737fb14f572f632f7b4")
+    version("0.1.16", sha256="49ced56992ba8be8aa6638525e8078b7e3ce0d5c05c34ee90746cab02bb5185a")
+    version("0.1.15", sha256="8c7c509db692b482cba5b0453579747db5a981ce5b3c13da96b14ae0332a6e81")
+    version("0.1.14", sha256="a5c75df1c3ef6fac10d92fb6781643e0834e5c35debe77693686dab8bfcf221f")
+    version("0.1.13", sha256="8263938e49b501c477f626b4c25e0c74e91152268830c69aabc96eeb263c6eea")
+    version("0.1.12", sha256="f0fa0f3b129d28e41b337ce2c39c3604990752de8e485327ec9df3bf0360e9c1")
+    version("0.1.11", sha256="95f302818971fec3f19ef18febd5c31c580490692138c8e4fe3534104d88b5e0")
+    version("0.1.10", sha256="7ef9f911f7ea31da5ff5306d8372ec194d223850aede0878ac2a921ce049bbb2")
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/py-libsonata/package.py:36]: Are these older versions really needed? For a new package, usually the newer versions should be enough, or not in this case?

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    depends_on("cmake@3.3:", type="build")
+    depends_on("hdf5")
+    depends_on("py-pybind11")
+
+    depends_on("py-numpy@1.17:", type=("build", "run"))
+    depends_on("py-setuptools", type="build", when="@0.1:")
+    depends_on("py-setuptools-scm", type="build", when="@0.1:")
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/py-libsonata/package.py:44]: Here the minimum versions of those would also be good.

File: var/spack/repos/builtin/packages/py-pytest-mpi/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    pypi = "pytest-mpi/pytest-mpi-0.6.tar.gz"
+
+    license("BSD-3-Clause")
+
+    maintainers("tristan0x")
+
+    version("0.6", sha256="09b3cd3511f8f3cd4d205f54d4a7223724fed0ab68b872ed1123d312152325a9")
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pytest", type=("build", "run"))
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/py-pytest-mpi/package.py:21]: Likewise, please add the minimum versions or those that work for sure.

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

**Review by bernhardkaindl — COMMENTED** [review]
Hello @tristan0x, We value your contribution and would of course be happy to merge these packages, and while I do personally think that having packages is better than having them not is good, a few of these packages are missing some minimum versions for their dependencies.

Usually it is far easier on everyone, even if you like to build a stack of packages where one package depends on the other before it in the stack, to open a new pull request, and start with the package at the bottom of the stack. Then that has been merged, progress to the next.

I've set this to back to draft to stop GitLab CI for it because even if you'd not want to go for an initial small PR that only adds the first package in the stack or chain, this will for sure receive other review comments, and you'd have to make changes anyway that would need to re-run CI.

@tristan0x I don't mean this to throw you back, and this is my feedback from what I've learnt! I might be wrong though!

However, independent of whether my feedback is correct or not, I think it's always easier to get new packages off the ground by starting as small and easy as possible.

**Comment by tristan0x** [comment]
Thank you for your feedback @bernhardkaindl. I am closing this PR to split this contribution.
